### PR TITLE
Openssl 3.0: RSA prep work

### PR DESCRIPTION
**Pull request by fredmorcos**
### Short description
Easy stuff, first commit is formatting (automated) and second commit is lint fixes (mostly automated). I suggest to review commit by commit.

### Checklist
I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by fredmorcos** [comment]
Actually, I'd like to get #11600 merged first then I'll rebase this PR.

**Review by rgacogne — APPROVED** [review]
Looks good to me, nice cleanup! I made a few suggestions but nothing that would prevent a merge as far as I am concerned.

File: pdns/opensslsigners.cc [hunk 1]
```diff
@@ -382,121 +399,123 @@ int OpenSSLRSADNSCryptoKeyEngine::hashSizeToKind(const size_t hashSize)
...
 
-  res = SHA1_Final(l_hash, &ctx);
+  res = SHA1_Final(l_hash.data(), &ctx);
   if (res != 1) {
-    throw runtime_error(getName()+" failed to finish hash context for generating the public key hash");
+    throw runtime_error(getName() + " failed to finish hash context for generating the public key hash");
   }
 
-  return string((char*)l_hash, sizeof(l_hash));
+  return {(char*)l_hash.data(), sizeof(l_hash)};
```
[the remark this answers is not in the document]
Comment (fredmorcos) [reply pdns/opensslsigners.cc:464]: Done.

**Review by omoerbeek — APPROVED** [review]

**Comment by fredmorcos** [comment]
The new changes are just due to the rebase on master.

**Review by rgacogne — APPROVED** [review]

**Comment by rgacogne** [comment]
First failure of `Build and test everything / test-dnsdist-regression (tsan)` was unrelated to this PR, it looks like the first query was sent when dnsdist was not yet ready.

**Pull request merged**